README.md
# MagicConch

<img src="https://github.com/NestIntro/MagicConch/assets/91196025/316b32b3-de60-4003-8d76-c316cd711f69" width="170px" height="170px" align="center"/>

<br/>

## 🐚 What is MagicConch?

If you have an unsolvable problem, ask the magic conch

<br/>

## 🧙 Team

### GP(Good People) & GP(Good Project)

<br/>
<table>
  <tbody>
    <tr>
      <td align="center"><a href=""><img src="https://file3.instiz.net/data/cached_img/upload/2021/03/02/20/d9459e5a8b7552eb00b5a5f3bb9f517a.jpg" width="150px"  height="150px"; alt=""/><br /><sub><b>김지환 </b></sub></a><br /></td>
      <td align="center"><a href=""><img src="https://i.pinimg.com/736x/a3/fe/d2/a3fed2746329f5dfcc3847ac199bba48.jpg" width="150px" height="150px" alt=""/><br /><sub><b>남효정 </b></sub></a><br /></td>
      <td align="center"><a href=""><img src="https://cdn.indiepost.co.kr/uploads/images/2018/12/11/uDZ4Pt-700x525.png" width="150px" height="150px" alt=""/><br /><sub><b>이다정</b></sub></a><br /></td>
      <td align="center"><a href=""><img src="https://mblogthumb-phinf.pstatic.net/MjAxODA1MTRfMTA0/MDAxNTI2MjI0NTA0MDAy.8QZYsuFXxoLDLIt5WsV0NQG9h46BDkya-Dyi4mCxoi0g.yeG1Lzw30K6Dx-sEBjSOOKrnBmLk9Q60xXQk3IBeN0kg.PNG.wnxor520/20180513_233340.png?type=w800" width="150px" height="150px" alt=""/><br /><sub><b>최희원</b></sub></a><br /></td>
    </tr>
  </tbody>
</table>
<br/>

## 💻 Stack

<div>
  <img src="https://img.shields.io/badge/JAVASCRIPT-F7DF1E?style=for-the-badge&logo=JavaScript&logoColor=white">
  <img src="https://img.shields.io/badge/NODE.JS-339933?style=for-the-badge&logo=Node.js&logoColor=white">
  <img src="https://img.shields.io/badge/NPM-CB3837?style=for-the-badge&logo=npm&logoColor=white">
  <img src="https://img.shields.io/badge/GITHUB-181717?style=for-the-badge&logo=GitHub&logoColor=white">
  <img src="https://img.shields.io/badge/WEBPACK-8DD6F9?style=for-the-badge&logo=Webpack&logoColor=white">
  <img src="https://img.shields.io/badge/.ENV-ECD53F?style=for-the-badge&logo=.ENV&logoColor=white">
</div>

<br/>

## Demo
<img src="https://github.com/NestIntro/MagicConch/assets/91196025/3a5828d2-3aef-4ea1-bca8-e1d9a65c17f1" width="600px" height="270px"/>

<br/>

## Getting Started

```
npm i magic-conch-soragodong
```
<br/>

## Usage

```
node node_modules/magic-conch-soragodong/cli.js
```

<br/>

## Contributing

- [Code of Conduct](https://www.contributor-covenant.org/)
<br/>
- [Issue guideline](https://github.com/NestIntro/MagicConch/tree/main/.github/ISSUE_TEMPLATE)

<br/>

## :loudspeaker: Commit Message Conventions

## Commit Message Format

It is consist of subject, body, and footer. 

Each part is separated by a single line.


```markdown
<Commit Type>: subject // 📝 you can describe shortly about your commit

body // 📝 if you have to describe more detaily what you did, you can write down on this part. It is optional

footer // 📝 this part exists for tracking issue.
```
<br/>
📝 Here is an example

```markdown
[Feat]: Add an authentication function   // subject part

User authentication was implemented using the Jwt token method // body part

Resolves: #18   // footer part
```

<br/>

## Commit Type

Must be one of the following:

- **Feat**: A new feature
- **Fix**: A bug fix
- **Docs**: Documentation only changes
- **Style**: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- **Refactor**: A code change that neither fixes a bug nor adds a feature
- **Test**: Adding missing or correcting existing tests
- **Chore**: Changes to the build process or auxiliary tools and libraries such as documentation generation
- **Design**: Changes the Ui design (CSS, etc)
- **Comment**: Changes comments
- **Init**: Initial project setup
- **Rename**: Changes file/folder name or the path
- **Remove**: Files only remove

<br/>

## Subject Rule

- Subject cannot be longer 50 characters.
- don't use dot (.) at the end and special symbols.
- use the imperative, **present** tense: "change" not "changed" nor "changes"
- capitalize the first letter

<br/>

## Body Rule

- Basically, body can be written selectively.
  - 🤔 However, if it is not possible to explain simply with a subject, it is better to write it.
- The content of the text should be written in as much detail as possible regardless of the amount, but it should not exceed 80 characters.
- The content of the text must include an explanation of what was changed and why.

<br/>

## Footer Rule

```
[Issue tracker type]:#[Issue number]
ex-1). Resolves:#17
ex-2). Fixes: #17 Related to: #12, #16
```

- Footer is phrases for issue tracking.
  - 🤔 If it is not a commit linked to the issue, you can selectively write it.
- Specify the type of issue.
- When writing down multiple issue numbers, separate them with commas ().

<br/>

## Issue tracker type

Must be one of the following:

- **Fixes**: Fix an issue that has not yet been resolved
- **Resolves**: Issue resolved
- **Ref**: Issue exists for reference
- **Related to**: Unresolved issues associated with this commit

<br/>
## 📜 License

This software is provided under [MIT License](https://github.com/nhn/toast-ui.doc/blob/master/LICENSE).


CHANGELOG.md
## [1.0.8](https://github.com/NestIntro/MagicConch/compare/v1.0.7...v1.0.8) (2023-08-19)



## [1.0.7](https://github.com/NestIntro/MagicConch/compare/v1.0.6...v1.0.7) (2023-08-18)



## [1.0.6](https://github.com/NestIntro/MagicConch/compare/v1.0.5...v1.0.6) (2023-08-16)



## [1.0.5](https://github.com/NestIntro/MagicConch/compare/v1.0.4-beta...v1.0.5) (2023-08-16)



## [1.0.4](https://github.com/NestIntro/MagicConch/compare/v1.0.4-beta...v1.0.4) (2023-08-16)



## [1.0.3](https://github.com/NestIntro/MagicConch/compare/v1.0.4-beta...v1.0.3) (2023-08-16)



# [1.0.0](https://github.com/NestIntro/MagicConch/compare/v1.0.4-beta...v1.0.0) (2023-08-16)



## [1.0.4-beta](https://github.com/NestIntro/MagicConch/compare/v1.0.2-beta...v1.0.4-beta) (2023-08-16)



## [1.0.3-beta](https://github.com/NestIntro/MagicConch/compare/v1.0.2-beta...v1.0.3-beta) (2023-08-16)



## [1.0.2-beta](https://github.com/NestIntro/MagicConch/compare/v1.0.1-beta...v1.0.2-beta) (2023-08-15)



## 1.0.1-beta (2023-08-15)



# 1.0.0-beta (2023-08-15)



## 1.0.2 (2023-08-15)



## 1.0.3 (2023-08-15)



## 1.0.11 (2023-08-15)



## 1.0.10 (2023-08-15)



## 1.0.9 (2023-08-14)



## 1.0.8 (2023-08-14)



## 1.0.7 (2023-08-14)



## 1.0.6 (2023-08-14)



## 1.0.5 (2023-08-11)





magic.js
import * as readline from "readline";
import chalkRainbow from "chalk-rainbow";

const answerlist = {
  answers: {
    0: "어",
    1: "아니",
    2: "싫어",
    3: "그래",
    4: "좋아",
    5: "돼",
    6: "안돼",
    7: "생각하고 말한 거야?",
    8: "충분해",
    9: "당장 해",
    10: "언젠가는",
    11: "되겠냐 그게",
    12: "오 그런 것도 질문이구나",
    13: "당연하지",
    14: "안알랴줌ㅋ",
    15: "흠",
    16: "오 나도 궁금했던건데",
    17: "내일까지 알아와",
    18: "이젠 좀 알 때도 되지 않았나?",
    19: "맞아",
    20: "졸리니까 내일 알려줄게",
    21: "내가 어제 알려줬잖아",
    22: "어이쿠 저런!",
    23: "안타깝네..",
    24: "오 대박",
    25: "사람은 좋은데,, 질문이 참,,",
    26: "너가 나보다 낫다",
    27: "한국말로 해",
    28: "크.. 그거지",
    29: "뭘 좀 아네",
    30: "너 T지?",
    31: "역시~!",
    32: "귀욥네",
    33: "ㅋ",
  },
};

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

export default function give_me_answer() {
  rl.setPrompt("> ");
  rl.prompt();
  rl.on("line", function (line) {
    switch (line) {
      case "quit":
        rl.close();

      default:
        console.log(
          "🐚 :",
          chalkRainbow(answerlist.answers[Math.floor(Math.random() * 32 + 1)])
        );
        rl.prompt();
    }
  });
  rl.on("close", function () {
    process.exit();
  });
}

// give_me_answer();
